Add include_media FormHelper attribute to make form media inclusion optional

abbottc opened the pull request:
This adds a FormHelper attribute "include_media", which defaults to True and controls whether form media is included with the form. Having this attribute makes it much easier to use crispy forms in scenarios which require special handling of media, such as with javascript loading systems, with django-compressor (in offline compression mode), and when the media might produce duplicate javascript when more than one form is on a page. Without this setting, crispy templates must be overridden, which is somewhat more cumbersome. The pull request includes code changes, documentation changes, and 6 new tests.

abbottc commented:
I just took a look at the continuous integration failure, and from the traceback it looks like the error occurred while pip was attempting to install the requirements for setting up the environment under Python 3.3 with Django 1.8 (before any tests got a chance to run for that environment).  I don't believe this represents an error with the pull request, but let me know if you come to a different conclusion.

carltongibson commented:
@abbottc — Thanks for this. I'll review ASAP. (First glance looks great though.)

I restarted the failed build — All green now.

carltongibson commented:
@abbottc This is great. If you could rebase I'll merge. Thanks for the effort!

abbottc commented:
I see that bootstrap4 support has been added to the dev brach, so I added the changes for those new templates to support the include_media attribute, and added two tests for the bootstrap4 templates and the attribute.

carltongibson commented:
Awesome. Excellent work. Let's get this in before something else changes.
